datafusion/expr-common/src/statistics.rs

```diff
@@ -0,0 +1,1610 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::f64::consts::LN_2;
+
+use crate::interval_arithmetic::{apply_operator, Interval};
+use crate::operator::Operator;
+use crate::type_coercion::binary::binary_numeric_coercion;
+
+use arrow::array::ArrowNativeTypeOp;
+use arrow::datatypes::DataType;
+use datafusion_common::rounding::alter_fp_rounding_mode;
+use datafusion_common::{internal_err, not_impl_err, Result, ScalarValue};
+
+/// New, enhanced `Statistics` definition, represents five core statistical
+/// distributions. New variants will be added over time.
+#[derive(Clone, Debug, PartialEq)]
+pub enum StatisticsV2 {
+    Uniform(UniformDistribution),
+    Exponential(ExponentialDistribution),
+    Gaussian(GaussianDistribution),
+    Bernoulli(BernoulliDistribution),
+    Unknown(UnknownDistribution),
```

Comment:
Unknwon name might be misleading, it looks like based on `mean, avg, variance`
```
UnknownDistribution::try_new(mean, median, variance, range).map(Unknown)
```
I would think unknown is no distribution or not usable one, but here likely it is usable

Reply:
Unknown distribution is indeed usable. It just means that we don't have exact information about how the random variable is distributed, and we are resorting to summary statistics to describe a generic distribution about which we have no special knowledge.

Reply:
If the naming trips up more people, maybe we can change it to something like `GenericDistribution`. I guess we will have enough eyes on this to know before we make a release with this code.

Reply:
Like it, `GenericDistribution` or maybe `MetricsDistribution`, or `AvgDistribution` to show the user the inner nature of this sort of distribution